core/font.lua

```diff
@@ -3,7 +3,10 @@ local icu = require("justenoughicu")
 local lastshaper
 
 SILE.registerCommand("font", function (options, content)
-  if SU.hasContent(content) then SILE.settings:pushState() end
+  if SU.hasContent(content) then
+    SILE.typesetter:initline()
```

Comment:
It will only work for the `\font` command, but not generally for all cases where one can enter a temporary settings wrapper. It seems an ad hoc workaround for but one case only, I don't know but it feels kind of contrived. As I wrote in the issue, `\set[parameter=font.style, value=italic]{content}` would still fail. Admittedly, changing font this way is not very clever, but that would be true for any other temporary setting. E.g. the following fails similarly:

```
\section{fail}

\language[main=fr]{doh} b c

d

e

\section{expected}

a b c

d

e
```

![image](https://user-images.githubusercontent.com/18075640/228400755-673fd811-04fc-4297-b522-e46a5a6a1a97.png)

In other terms, doing this in `\font` seems wrong to me. And cramming it into `SILE.settings:pushState()` (to get our back covered here, but also at every place where settings are explicitly pushed, or implicitly via a `SILE.settings:temporarily` wrapper), besides not really being doable, would introduce a severe separation of concerns issue...

The scoping issue is hard here. My gut feeling is that current.parindent is not a setting comparable to language or font changes, but a typesetter state.

Reply:
All true, this is only a band-aid on what is probably the most common iteration of a bigger problem. It's whacking off the visible top of the iceberg without doing anything about the floating mass, and something else is going to float up after this.

That being said I think the band-aid is better than the unexpected results in this common case and I expect a it will be *forced* into a better resolution when we have a settings stack that can be managed in the scope of a typesetter.

Reply:
I'd prefer working towards an actual fix rather than inserting a _ad hoc_ workaround for one case only, which make it easy to forget about it and later face other concerns. I'm not sure what the actual fix would be -- i.e. not sure we even need a "settings stack" for typesetter-scoped parameters (The typesetter already has a stack with states, and this could be sufficient).

Reply:
The typesetter has a state-stack, but typesetter related settings are not tracked in the "state" at all. That's part of our problem. A real fix for this is already tracked in other issues, namely #1327 and friends.

Reply:
> A real fix for this is already tracked in other issues, namely #1327

Exactly! So perhaps it's premature to add a "workaround" of sorts here for #1699 and turn to the real thing...
But we have to consider and better define what are states vs. settings. IMHO, not all settings called "typesetter.xxx" need to be typesetter _states_, and some "yyy.param" are actually typesetter states. -- `current.parindent` could _possibly_ be one. 

I'd like to suggest a path towards it at some point, but it likely requires thinking out-of-the-box i.e. attempting to forget the current implementation and re-think it from scratch (even if ending very similar eventually), so as to be sure asking the right questions.

Reply:
I have been attempting a "hack" to properly "scope" the parindent. It shows other things possibly done wrong (or wrong assumptions relying on side effects, e.g. in the book class -- I don't have the same issue in resilient.book at the same places)... And there are still a few failing non-regression tests that need analysis. I can try to clean that up and make a draft PR. It's probably not the _right_ way to go obviously but it does show something about proper scoping and may help understanding the issue at stakes.

Reply:
... And it helps finding tests that were likely wrong, e.g. `tests/chapterverse.sil` seems affected too by #1699 (no chapter opening, but temporary closure in header sets current.parindent to 0, yet a parindent is applied -- and as in the issue, this is the scoping in a `\font`...)

![image](https://user-images.githubusercontent.com/18075640/229969507-d4d7de34-451b-4802-ae74-9d0033202f0d.png)

Reply:
See #1760

Reply:
Additionally, regarding the initial proposal here, invoking `SILE.typesetter:initline()` outside the temporary environment will imply using the parindent from the upper context, which might be wrong if it is set to a font-relative unit, no?